hack/update-estimator-protobuf.sh

```diff
@@ -20,9 +20,26 @@ set -o pipefail
 
 KARMADA_ROOT="$(cd "$(dirname "${BASH_SOURCE[0]}")/.." && pwd -P)"
 
+# go-to-protobuf has a argument:
+#    -o, --output-base string        Output base; defaults to $GOPATH/src/
+# So the protobuf code generated to ${GOPATH}/src/github.com/karmada-io/karmada by default
+# But we usually fork the karmada repo from karmada-io/karmada, we may not put the root path to ${GOPATH}/src/github.com/karmada-io/karmada in local file system
+# To ensure that the generated file is under ${KARMADA_ROOT}, we create a temporary ${GOPATH} and soft link ${KARMADA_ROOT} to ${GOPATH}/src/github.com/karmada-io/karmada
+
+go_path="${KARMADA_ROOT}/_go"
+clean_gopath() {
+  # ${GOPATH}/pkg/mod directory is readonly, without go clean -modcache, rm command will fail with `Permission denied`
+  go clean -modcache
```

Comment:
We don't need to download mod files in this tmp path.

See https://github.com/karmada-io/karmada/blob/181af49291b25f3f710c645afec9683456dead3b/hack/update-codegen.sh#L35-L45.